console_backend/src/tabs/settings_tab.rs

```diff
@@ -599,18 +602,17 @@ impl SettingsTab {
         let original = self
             .settings
             .lock()
-            .get(group, name)?
-            .value
-            .as_ref()
-            .ok_or_else(|| anyhow!("{group}.{name} was none"))?
-            .clone();
+            .get(group, name)
+            .map(|x| x.value.clone())
+            .unwrap_or_default();
+        // if is in new config
         let in_config = conf
             .section(Some(group))
             .and_then(|s| s.get(name))
             .is_some();
         self.write_setting(group, name, value)?;
         if !in_config {
-            Ok(Some(original))
+            Ok(original) // revert to original if not in new config
```

Comment:
How does this handle the case where there is no original and it's not in the group, does it set the values to the default?

Reply:
when theres no original, it wont revert back to original i.e. it gets set
since we can't fetch the field.

this was because ntrip.enable wasn't found - which would turn ntrip to false by default (im assuming if its not found, it was false?)